content/bn/docs/concepts/architecture/garbage-collection.md

```diff
@@ -0,0 +1,136 @@
+
+v1.20+ এ, যদি গার্বেজ কালেক্টর একটি অবৈধ ক্রস-নেমস্পেস `ownerReference` সনাক্ত করে, অথবা একটি ক্লাস্টার-স্কোপড নির্ভরশীল একটি নেমস্পেসড প্রকারের মালিককে রেফারেন্স করে, তবে একটি সতর্কতা ইভেন্ট যার কারণে `OwnerRefInvalidNamespace` এবং একটি `involvedObject` থাকে যা অবৈধ নির্ভরশীল হিসেবে রিপোর্ট হয়। এধরনের ইভেন্ট দেখতে আপনি চালাতে পারেন kubectl get events `kubectl get events -A --field-selector=reason=OwnerRefInvalidNamespace`।
+{{< /note >}}
+
+## ক্যাসকেডিং ডিলিশন {#cascading-deletion}
+
+Kubernetes  সেসব অবজেক্ট পরীক্ষা করে এবং মুছে ফেলে, যেগুলোর আর মালিকানা রেফারেন্স নেই, যেমন `ReplicaSet` মুছে ফেললে বাকি থাকা পডগুলো। যখন আপনি একটি অবজেক্ট মুছে ফেলেন, তখন আপনি নিয়ন্ত্রণ করতে পারেন যে কুবার্নেটিস ঐ অবজেক্টের ডিপেনডেন্টসগুলো স্বয়ংক্রিয়ভাবে মুছে ফেলবে কিনা, এই প্রক্রিয়াকে বলা হয় *কাসকেডিং ডিলিশন*। কাসকেডিং ডিলিশনের দুটি ধরনের থাকে, যা নিচে ব্যাখ্যা করা হয়েছে:
+
+* ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশন
+* ব্যাকগ্রাউন্ড ক্যাসকেডিং ডিলিশন
+
+আপনি গার্বেজ কালেকশন কিভাবে এবং কখন মালিকানা রেফারেন্সসমূহের মাধ্যমে রিসোর্স মুছে ফেলে Kubernetes {{<glossary_tooltip text="Finalizers" term_id="finalizer">}} ব্যবহার করে নিয়ন্ত্রণ করতে পারেন।
+
+
+### ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশন {#foreground-deletion}
+
+ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশনে, আপনি যে মালিকানা অবজেক্টটি মুছে ফেলছেন সেটি প্রথমে একটি *ডিলিশন ইন প্রগ্রেস* অবস্থায় চলে যায়। এই অবস্থায়, মালিকানা অবজেক্টের সাথে নিম্নলিখিত ঘটনা ঘটে:
+
+* Kubernetes API সার্ভার অবজেক্টের `metadata.deletionTimestamp` ক্ষেত্রটিকে সেই সময়ে সেট করে যখন অবজেক্টটি মুছে ফেলার জন্য চিহ্নিত করা হয়।
+* Kubernetes API সার্ভার এছাড়াও `metadata.finalizers` ক্ষেত্রটিকে `foregroundDeletion` এ সেট করে।
+* অবজেক্টটি Kubernetes API-এর মাধ্যমে দৃশ্যমান থাকে যতক্ষণ না মুছে ফেলার প্রক্রিয়া সম্পূর্ণ হয়।
+
+মালিকানা অবজেক্টটি ডিলিশন ইন প্রগ্রেস অবস্থায় চলে গেলে, কন্ট্রোলার নির্ভরশীল অবজেক্টগুলি মুছে ফেলে। সমস্ত নির্ভরশীল অবজেক্ট মুছে ফেলার পর কন্ট্রোলার মালিকানা অবজেক্টটি মুছে ফেলবে। এই মুহূর্তে, অবজেক্টটি Kubernetes API-এ আর দৃশ্যমান থাকে না।
+
+ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশন চলাকালীন, একমাত্র নির্ভরশীলরা যারা মালিক মুছে ফেলার প্রতিবন্ধকতা সৃষ্টি করে তা হল যাদের `ownerReference.blockOwnerDeletion=true` ক্ষেত্র রয়েছে। আরো জানার জন্য [ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশন ব্যবহার করুন।](/bn/docs/tasks/administer-cluster/use-cascading-deletion/#use-foreground-cascading-deletion)
+
+### ব্যাকগ্রাউন্ড ক্যাসকেডিং ডিলিশন {#background-deletion}
+
+ব্যাকগ্রাউন্ড ক্যাসকেডিং ডিলিশনে, Kubernetes API সার্ভার অবজেক্টটি অবিলম্বে মুছে ফেলে এবং কন্ট্রোলার ব্যাকগ্রাউন্ডে নির্ভরশীল অবজেক্টগুলি পরিষ্কার করে। স্বাভাবিকভাবে, Kubernetes ব্যাকগ্রাউন্ড ক্যাসকেডিং ডিলিশন ব্যবহার করে যতক্ষণ না আপনি ম্যানুয়ালি ফরগ্রাউন্ড ডিলিশন ব্যবহার করেন বা নির্ভরশীল অবজেক্টগুলি অরফান না করেন।
+
+আরো জানার জন্য [ব্যাকগ্রাউন্ড ক্যাসকেডিং ডিলিশন ব্যবহার করুন।](/bn/docs/tasks/administer-cluster/use-cascading-deletion/#use-background-cascading-deletion)
```

Comment:
Use the English version as reference `[ব্যাকগ্রাউন্ড ক্যাসকেডিং ডিলিশন ব্যবহার করুন](/docs/tasks/administer-cluster/use-cascading-deletion/#use-background-cascading-deletion)।`